ksqldb-engine/src/test/java/io/confluent/ksql/function/udaf/offset/EarliestByOffsetUdafTest.java

```diff
@@ -55,6 +70,29 @@ public void shouldComputeEarliestInteger() {
 
     // Then:
     assertThat(res.get(VAL_FIELD), is(321));
+  }
+  
+  @Test
+  public void shouldComputeEarliest2Integers() {
+    // Given:
+    final Udaf<Integer, List<Struct>, List<Integer>> udaf = EarliestByOffset.earliestInteger(2);
+
+    // When:
+    List<Struct> res = udaf
+        .aggregate(123, new ArrayList<>(Arrays.asList(EarliestByOffset.createStruct(STRUCT_INTEGER, 321))));
+
+    // Then:
+    assertThat(res.get(0).get(VAL_FIELD), is(321));
+    assertThat(res.get(1).get(VAL_FIELD), is(123));
+    
+    List<Struct> res2 = udaf
+        .aggregate(543, res);
+    assertThat(res2.size(), is(2));
+    assertThat(res2.get(0).get(VAL_FIELD), is(321));
+    assertThat(res2.get(1).get(VAL_FIELD), is(123));
+    
+    
+    
```

Comment:
Probably better structured as two tests:

```suggestion
@Test
  public void shouldCaptureValuesUpToN() {
    // Given:
    final Udaf<Integer, List<Struct>, List<Integer>> udaf = EarliestByOffset.earliestInteger(2);

    // When:
    List<Struct> res0 = udaf.aggregate(321, new ArrayList<>());
    List<Struct> res1 = udaf.aggregate(123, res0);

    assertThat(res1, hasSize(2));
    assertThat(res1.get(0).get(VAL_FIELD), is(321));
    assertThat(res1.get(1).get(VAL_FIELD), is(123));
  }

  @Test
  public void shouldIgnoreValuesPastN() {
    // Given:
    final Udaf<Integer, List<Struct>, List<Integer>> udaf = EarliestByOffset.earliestInteger(2);

    final List<Struct> aggregate = Arrays.asList(
        EarliestByOffset.createStruct(STRUCT_INTEGER, 10),
        EarliestByOffset.createStruct(STRUCT_INTEGER, 3)
    );

    // When:
    final List<Struct> result = udaf.aggregate(2, aggregate);

    assertThat(result, hasSize(2));
    assertThat(result.get(0).get(VAL_FIELD), is(10));
    assertThat(result.get(1).get(VAL_FIELD), is(3));
  }```